Update data_editor docstring re: supported types

MathCatsAnd opened the pull request:
Added a sentence to `experimental_data_editor` docstring to clarify what data types are not supported for editing. 

Closes streamlit/docs#636

**Contribution License Agreement**
By submitting this pull request you agree that all contributions to this project are made under the Apache 2.0 license.

MathCatsAnd committed:
Update data_editor docstring re: supported types

Added sentence to docstring to clarify what data types are not supported for editing.

MathCatsAnd committed:
Edit data_editor docstring re: supported types

Expanded verbage to clarify type of data within a data structure vs the type of the enveloping data structure itself.

snehankekre committed:
Lint with Black to fix failing CI tests

lukasmasuch commented:
@MathCatsAnd  Thanks for adding this 🙏 there are a few Pandas specific types which aren't yet editable as well:  `pd.Period` (period), `pd.Interval` (interval), `pd.Timedelta` / `datetime.timedelta` (timedelta) and we should probably also add `decimal.Decimal` (decimal) here.. maybe also `fractions.Fraction` but this one is super uncommen. Question is if we add them with the real object name (`pd.Interval`) or just with a shortened version (`interval`)? Any preference @jrieke / @snehankekre?

MathCatsAnd committed:
Merge branch 'streamlit:develop' into data-editor-supported-types

MathCatsAnd commented:
I'm writing up an edit and including the module name consistent with how they are used in the docstring to describe the parameters, but feel free to tell me otherwise. ~~I could not find a pandas.Decimal. Do you mean pyarrow.decimal128? I've never dealt directly with pyarrow myself so I'm not sure how that type could come into play implicitly through pandas. I looked here: https://pandas.pydata.org/docs/reference/arrays.html~~ I misread. Fixing it now.

MathCatsAnd committed:
Added pandas data types not supported by data editor

Expanded `st.experimental_data_editor` docstring warning about unsupported types. Pandas and pyarrow upsupported types now included.

MathCatsAnd committed:
Corrected reference to unsupported decimal type

Removed pyarrow.decimal128 from unsupported list and replaced with intended decimal.Decimal

lukasmasuch reviewed (approved):
LGTM 👍

lukasmasuch commented:
@MathCatsAnd one last thing we might want to add here: mixing types (e.g. string & int) in a column will also result in a non-editable column in most cases. We could just add something like `mixed types` here, or maybe attach an additional sentence:
Mixing different types within a single column can make it uneditable.

MathCatsAnd committed:
Warning about mixing data types within a column of the data editor

Added a warning that mixing data types within a column of the data editor can make the column uneditable.

MathCatsAnd committed:
Rearranged data editor type restrictions for better typesetting

Combined notes into a single paragraph for better display in documentation.

snehankekre commented:
> add them with the real object name (pd.Interval) or just with a shortened version (interval)? Any preference

Real object name should cause less ambiguity 👍
